### PR TITLE
Jobs queue refactoring for url-analyzer

### PR DESCRIPTION
**Pull request by jeroenpf**
This PR is part of a series of PR's related to adding URL analyzer capabilities to the mShot service. (see #53)

Particularly, this PR is introducing some refactoring of the way jobs are queued, deleted, and accessed. Mainly to prevent adding more conditions to parts of the code later.

Previously, jobs were identified by a number of variables. Together, these variables would make a job 'unique' to prevent duplicate jobs and to retrieve jobs from the queue and eventually delete them. To make things a bit easier, i've added a unique key for each job.  Specifically for deleting jobs, there was a lot of logic that has been simplified instead of adding more if-else statements to deal with url-analyzer jobs later.

Unit tests can be used to verify the refactoring does not break existing functionality. Also, it is good to try and capture a screenshot and see if it produces the expected result.

### CODE REVIEW

File: lib/common.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
+function delete_from_queue( job_key, status ) {
```
Comment (vishnugopal) [inline lib/common.js:1]: Can we get some JSDOC for these functions please?

**Review by vishnugopal — APPROVED** [review]
The functionality continues to work, and the code looks much neater!

@zaerl Please also take a look at this to see if it makes sense.

**Comment by zaerl** [comment]
@vishnugopal the code is ok!

**Review by zaerl — APPROVED** [review]
It's ok!

**Pull request merged**